### DISCUSSION TITLE
xDrip+ not following via mobile data

### DISCUSSION THREAD

**Discussion by mlkoldau** [NightscoutFoundation/xDrip#2150] [General]
Hello :-) I am using my Samsung Galaxy S6 (SM-G920I, running Android 7.0, everything except xDrip+ works fine) to follow as xDrip+ Sync Follower the xDrip+ (with Dexcom Anubis G6) of my T1D daughter.
When my phone is accessing the Internet via WiFi xDrip+ is working great, but if I'm away from home and there is no WiFi and the phone has to use mobile data, my xDrip+ does not get data. Internet browser and Google and all other internet-using apps work fine on mobile data, just not xDrip+.
In the event log it shows occasionally messages like (not every 5min) "DoNothingService 12:23 Restarting collection service + full wakeup due to minsago:63!!!", nothing else what could be helpful.
If I reboot the phone I get one dot, then it does not receive any further data. If I reboot again I get one dot again...
I checked all the usual suspects like "Allow background data usage", "Allow unrestricted data usage", "Do not optimise battery usage", permissions for everything, ..., but nothing stands out.
My wife is following my daughter's xDrip+ with an older Samsung J5 (SM-J500Y, Android 6.0.1) and she receives xDrip+ data without problems both via WiFi and mobile data.
We always use the latest "nightly" xDrip+ version.
Does anybody have some hints how to fix the problem? My old Samsung S6 is great except for that xDrip+ data problem :-(

**Comment by Navid200** [comment]
Would you please take a screenshot of the following page, blackout "Handset Group Security Sync Key", and post here?

Settings &#8722;> xDrip+ Sync settings

**Reply by mlkoldau** [reply]
It uses a very long security key, which (I believe) was suggested by the system as random code when we set the following up 2 or 3 years ago. I don't know if for somewhat reason my S6 could have problems with the long security key when using mobile data while my wifes Android 6 Samsung J5 does not have  a problem with this
![xDripPlus_SyncSettings](https://user-images.githubusercontent.com/106404107/170779038-08630216-cbcc-47aa-885a-5845978b1319.jpg)
.

**Reply by Navid200** [reply]
I'm curious, what you see if you tap on "Desert Sync".

**Comment by mlkoldau** [comment]
All of our phones have fixed IP addresses when on the home WiFi network and I have set up DesertSync because it seems that our internet connection can drop at night and by using desert sync it will carry on.
I switch the Desert Sync off every morning on my phone and switch it back on in the evening.
On my daughters phone desert sync is on all the time. Maybe that could cause the problems on my S6???

![Screenshot_20220528-080623 1](https://user-images.githubusercontent.com/106404107/170782777-c77e91e4-a297-4c4b-a351-3c3170e523a0.png)
![Screenshot_20220528-080636 1](https://user-images.githubusercontent.com/106404107/170782855-85391620-cb94-4135-9d2f-efccaf827489.png)

**Comment by Navid200** [comment]
@jamorham Would you please have a look at this?

**Comment by Navid200** [comment]
Have you disabled battery optimization of xDrip by Android?

**Reply by mlkoldau** [reply]
Battery optimization of xDrip by Android is disabled.

**Comment by Navid200** [comment]
In Android Settings, what do you see at the following?
Android Settings &#8722;> Apps & notifications &#8722;> See all apps &#8722;> xDrip+ &#8722;> Mobile data & WiFi

**Reply by mlkoldau** [reply]
![MobileDataSettings](https://user-images.githubusercontent.com/106404107/172335280-31066003-6909-47f7-a842-30efd0bd7036.png)
![MobileDataSettings](https://user-images.githubusercontent.com/106404107/172335364-9c21298c-1ad4-4600-b4b7-ea2734a17d62.png)

**Comment by mlkoldau** [comment]
Update:
I have switched Desert Sync this morning not only off on my S6 phone (follower), but also on my daughter's A31 phone (master). I have been outside around our farm this morning for quite a while (i.e. outside WiFi range) and my phone has kept receiving xDrip+ data!
So the tentative conclusion is:
My S6 does not receive xDrip+ data via mobile data, if Desert Sync on the master is turned on.
I will play with it and watch it over the next some days and report back.

**Comment by mlkoldau** [comment]
Update:
It seems to be that having Desert Sync off on the master (my daughter's Samsung A31) and on the follower (my S6) has somehow helped, but not solved the problem. If I'm outside and out of WiFi range, I sometimes have longer periods (say 30min) where xDrip+ is receiving data, but then it stops again, so problem not solved yet.
